File: Backend-Laravel/backend-TA/app/Models/OrderItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class OrderItem extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        "order_id",
        "products_id",
        "weight",
        "price",
        
    ];

    
    public function product()
    {
        return $this->belongsTo(Product::class, 'products_id');
    }

    public function order()
    {
        return $this->belongsTo(Order::class);
    }
}

File: Frontend/lib/database/order_item.dart
class OrderItem {
  final int? id;
  final int orderId;
  final int productId;
  final double weight;
  final int price;

  OrderItem({
    this.id,
    required this.orderId,
    required this.productId,
    required this.weight,
    required this.price,
  });

  OrderItem copyWith({
    int? id,
    int? orderId,
    int? productId,
    double? weight,
    int? price,
  }) {
    return OrderItem(
      id: id ?? this.id,
      orderId: orderId ?? this.orderId,
      productId: productId ?? this.productId,
      weight: weight ?? this.weight,
      price: price ?? this.price,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'order_id': orderId,
      'product_id': productId,
      'weight': weight,
      'price': price,
    };
  }
}
